test/extended/builds/build_timing.go

```diff
@@ -0,0 +1,102 @@
+package builds
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	"github.com/openshift/origin/pkg/build/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func verifyStages(stages []api.StageInfo, expectedStages map[string][]string) {
+
+	for _, stage := range stages {
+		expectedDurations, ok := expectedStages[string(stage.Name)]
+		if !ok {
+			fmt.Fprintf(g.GinkgoWriter, "Unexpected stage %v was encountered", stage.Name)
+			o.Expect(ok).To(o.BeTrue())
```

Comment:
when calling gomega from within a helper function, use ExpectWithOffset so the error reporting shows you who the actual caller.